apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditBuyAmount/SelectBuyTokenFromListScreen/SelectBuyTokenFromListScreen.tsx

```diff
@@ -190,7 +190,7 @@ const SelectableToken = ({tokenForList, wallet}: SelectableTokenProps) => {
       to_asset: [{asset_name: tokenForList.name, asset_ticker: tokenForList.ticker, policy_id: tokenForList.group}],
     })
     buyTouched()
-    buyAmountChanged({tokenId: tokenForList.id, quantity: Quantities.zero})
+    buyTokenIdChanged({tokenId: tokenForList.id, pools: []})
```

Comment:
buyTokenIdChanged should happen inside onSuccess of getting pools, so dispatch together the list, this will help removing some effects.

when changing will it be defaulted? then dispatch inside onsuccess too

Reply:
I get the intent of having pools as an argument of buyTokenIdChanged. But then what does the onSelect do? We fetch the pools with a react query hook that takes the ids as arguments, you can't really call that from here, you need to change the source of the ids that the hooks use.

Reply:
Never mind, I see by the later commits you already realized this and it's no longer an argument